 README.md           | 19 +++++++++++++++++++
 win32/Makefile.msvc | 29 +++++++++++++++++++++--------
 2 files changed, 40 insertions(+), 8 deletions(-)

diff --git a/README.md b/README.md
new file mode 100644
index 0000000..6dcb666
--- /dev/null
+++ b/README.md
@@ -0,0 +1,19 @@
+# libxml2
+
+libxml2 is the XML C parser and toolkit developed for the Gnome project from
+[http://www.xmlsoft.org/](http://www.xmlsoft.org/)
+
+PHP 5.3/5.4 currently use version 2.7.8 released 2010.11.04
+PHP 5.5 currently uses version 2.9.1 released 2013.04.19
+
+# Building for PHP with VC9/VC11
+
+Configure line non debug:
+
+    cscript configure.js lib="<path to iconv lib dir>" include="<path to iconv header dir>" vcmanifest=yes
+
+Configure line with debug:
+
+    cscript configure.js lib="<path to iconv lib dir>" include="<path to iconv header dir>"  debug=yes mem_debug=yes  vcmanifest=yes
+
+libxml2_a_dll.lib **MUST** be used with PHP.
\ No newline at end of file
diff --git a/win32/Makefile.msvc b/win32/Makefile.msvc
index 725b0ca..d6a381c 100644
--- a/win32/Makefile.msvc
+++ b/win32/Makefile.msvc
@@ -21,12 +21,21 @@ AUTOCONF = .\config.msvc
 # Names of various input and output components.
 XML_NAME = xml2
 XML_BASENAME = lib$(XML_NAME)
+!if "$(DEBUG)" == "1"
+XML_SO = $(XML_BASENAME)_debug.dll
+XML_IMP = $(XML_BASENAME)_debug.lib
+XML_A = $(XML_BASENAME)_a_debug.lib
+XML_A_DLL = $(XML_BASENAME)_a_dll_debug.lib
+XML_PDB = $(XML_BASENAME)_debug.pdb
+!else
 XML_SO = $(XML_BASENAME).dll
-XML_RES = $(XML_BASENAME).res
 XML_IMP = $(XML_BASENAME).lib
-XML_DEF = $(XML_BASENAME).def
 XML_A = $(XML_BASENAME)_a.lib
 XML_A_DLL = $(XML_BASENAME)_a_dll.lib
+XML_PDB = $(XML_BASENAME).pdb
+!endif
+XML_RES = $(XML_BASENAME).res
+XML_DEF = $(XML_BASENAME).def
 
 # Place where we let the compiler put its output.
 BINDIR = bin.msvc
@@ -67,8 +76,12 @@ LIBS =
 LIBS = $(LIBS) wsock32.lib ws2_32.lib
 !endif 
 !if "$(WITH_ICONV)" == "1"
-LIBS = $(LIBS) iconv.lib
-!endif 
+!if "$(DEBUG)" == "1"
+LIBS = $(LIBS) libiconv_debug.lib
+!else
+LIBS = $(LIBS) libiconv.lib
+!endif
+!endif
 !if "$(WITH_ICU)" == "1"
 !if "$(STATIC)" == "1"
 LIBS = $(LIBS) advapi32.lib sicuuc.lib sicuin.lib sicudt.lib
@@ -100,7 +113,7 @@ ARFLAGS = /nologo
 CFLAGS = $(CFLAGS) /D "_DEBUG" /Od /Z7
 LDFLAGS = $(LDFLAGS) /DEBUG
 !else
-CFLAGS = $(CFLAGS) /D "NDEBUG" /O2 
+CFLAGS = $(CFLAGS) /D "NDEBUG" /O2
 # commented out as this break VC10 c.f. 634846
 # LDFLAGS = $(LDFLAGS) /OPT:NOWIN98
 LDFLAGS = $(LDFLAGS)
@@ -362,15 +375,15 @@ $(XML_INTDIR_A_DLL) :
 
 # An implicit rule for libxml compilation.
 {$(XML_SRCDIR)}.c{$(XML_INTDIR)}.obj::
-	$(CC) $(CFLAGS) /Fo$(XML_INTDIR)\ /c $<
+	$(CC) $(CFLAGS) /Fo$(XML_INTDIR)\ /Zi /Fd$(BINDIR)\$(XML_PDB) /c $<
 
 # An implicit rule for static libxml compilation.
 {$(XML_SRCDIR)}.c{$(XML_INTDIR_A)}.obj::
-	$(CC) $(CFLAGS) /D "LIBXML_STATIC" /Fo$(XML_INTDIR_A)\ /c $<
+	$(CC) $(CFLAGS) /D "LIBXML_STATIC" /Fo$(XML_INTDIR_A)\ /Zi /Fd$(BINDIR)\$(XML_PDB) /c $<
 
 # An implicit rule for static for dll libxml compilation.
 {$(XML_SRCDIR)}.c{$(XML_INTDIR_A_DLL)}.obj::
-	$(CC) $(CFLAGS) /D "LIBXML_STATIC" /D "LIBXML_STATIC_FOR_DLL" /Fo$(XML_INTDIR_A_DLL)\ /c $<
+	$(CC) $(CFLAGS) /D "LIBXML_STATIC" /D "LIBXML_STATIC_FOR_DLL" /Fo$(XML_INTDIR_A_DLL)\ /Zi /Fd$(BINDIR)\$(XML_PDB) /c $<
 
 # Compiles libxml source. Uses the implicit rule for commands.
 $(XML_OBJS) : $(XML_INTDIR) 
